connectivity/check/test.go

```diff
@@ -390,6 +420,13 @@ func configureNamespaceInPolicySpec(spec *api.Rule, namespace string) {
 	}
 }
 
+// WithConditionFn takes a function containing condition check logic that
+// decides if the test needs to be run.
+func (t *Test) WithConditionFn(fn func() bool) *Test {
```

Comment:
let's call it `WithCondition`:
```suggestion
// WithCondition takes a function containing condition check logic that
// returns true if the test needs to be run, and false otherwise.
func (t *Test) WithCondition(fn func() bool) *Test {
```

Reply:
Fixed.